From 9ba8a54a3ae665ecee6d63ee8409a41334964684 Mon Sep 17 00:00:00 2001
From: Razvan A Lupusoru <razvan.a.lupusoru@intel.com>
Date: Mon, 3 Jun 2013 16:44:04 -0700
Subject: Dalvik: Fix libdvm_assert.so building

BZ: 113230

An assert condition performed an invalid operation which caused libdvm_assert
variant to not build.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-ME
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I6ec31c79843e37c45957dd9fd8b2eeea7e54d07f
Orig-MCG-Change-Id: Iac3684d3dc800a492fa7bded9fc52dbcd8b03a61
Signed-off-by: Razvan A Lupusoru <razvan.a.lupusoru@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/AccumulationSinking.cpp |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/vm/compiler/AccumulationSinking.cpp b/vm/compiler/AccumulationSinking.cpp
index e02721b..b1e2967 100644
--- a/vm/compiler/AccumulationSinking.cpp
+++ b/vm/compiler/AccumulationSinking.cpp
@@ -217,8 +217,8 @@ static bool checkNoOtherUses(CompilationUnit *cUnit, LoopInformation *info, unsi
 
     assert(defMIR->ssaRep);
     assert(defMIR->ssaRep->numDefs > 0);
-    assert(dvmExtractSSARegister(cUnit, defMIR->ssaRep->numDefs[0]) == vr ||
-            (defMIR->ssaRep->numDefs > 1 && dvmExtractSSARegister(cUnit, defMIR->ssaRep->numDefs[1]) == vr));
+    assert(dvmExtractSSARegister(cUnit, defMIR->ssaRep->defs[0]) == vr
+            || (defMIR->ssaRep->numDefs > 1 && dvmExtractSSARegister(cUnit, defMIR->ssaRep->defs[1]) == vr));
 
     //Get the SSA representation
     SSARepresentation *ssa = defMIR->ssaRep;
-- 
1.7.4.1

